examples/dataflow-hashpipeline/infrastructure/main.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import requests
+import json
+from datetime import datetime
+
+PROJECT         = os.getenv('PROJECT')
+REGION          = os.getenv('REGION')
+SECRET          = os.getenv("SECRET")
+BUCKET          = os.getenv('BUCKET')
+SERVICE_ACCOUNT = os.getenv('SERVICE_ACCOUNT')
+COLLECTION      = os.getenv('COLLECTION')
+TOPIC           = os.getenv('TOPIC')
+SALT            = os.getenv('SALT')
+
+def trigger_dataflow(data, context):
```

Comment:
This seems like a problematic pattern to me.

README indicates scalability is a goal of this asset.

A lot of big data delivery systems will dump a batch of files.
In this case you will trigger a dataflow pipeline per file (which will likely waste resources and be hard to monitor for a large batch of files).
If you expect files often you will be constantly spinning up and down pipelines.

This will also add huge latency hit (minutes for dataflow workers spin up) if you have near real time requirements.

Instead you can use TextIO to match files based on a gcs wildcard / prefix continuously in a dataflow pipeline in streaming mode.

This will also mean you can use the natively supported PubsubIO instead of your batch WriteToPubsub DoFn (which may become a maintenance burden).